src/identity/manager.rs

```diff
@@ -414,6 +415,20 @@ impl SecretManager<CaClient> {
     }
 }
 
+impl SecretManager<CustomSigner> {
+    pub async fn origin(cfg: crate::config::Config) -> Result<Self, anyhow::Error> {
+        let k8s_client = tls::create_k8s_client(cfg.k8s_ca_crt).await;
+        match k8s_client {
```

Comment:
This won't work on VMs. 

Pretty sure there is a way to initialize k8s client in rust that works with both 'in cluster' and kubeconfig. 

Same for the tokens - outside of K8S you need all the auth methods, the VM may use MDS or external commands to authenticate ( gcloud, etc ).